backend/src/api/routers/graphs.py

```diff
@@ -81,10 +81,40 @@ async def get_agent_response(user_input: UserInput) -> dict[str, Union[str, list
     else:
         raise ValueError('RetrieverGraph not initialized.')
 
-    tool = list(output)[0]['agent']['tools'][0]['name']
-    context = output[1][tool]['context']
-    sources = output[1][tool]['sources']
-    llm_response = output[2]['generate']['messages'][0]
+    if (
+        isinstance(output, list)
+        and len(output) > 0
+        and 'agent' in output[0]
+        and 'tools' in output[0]['agent']
+        and len(output[0]['agent']['tools']) > 0
+    ):
+        tool = output[0]['agent']['tools'][0].get('name', '')
+    else:
+        print('Tool name extraction failed')
+
+    if tool and isinstance(output, list) and len(output) > 1 and tool in output[1]:
+        if 'context' in output[1][tool]:
+            context = output[1][tool].get('context', '')
+        else:
+            print('Context extraction failed')
+
+        if 'sources' in output[1][tool]:
+            sources = output[1][tool].get('sources', '')
+        else:
+            print('Sources extraction failed')
```

Comment:
Is it fatal for the code if any source / context extraction failed?

Reply:
No, the sources/context sent back in the response body will be empty in that case.